src/components/Editor/index.tsx

```diff
@@ -211,6 +220,105 @@ const Editor = () => {
+        beginLineNumToLogEventNumMapRef:React.RefObject<BeginLineNumToLogEventNumMap>
+    ) => {
+        const selectedLogEventLineNum =
+            getMapKeyByValue(beginLineNumToLogEventNumMapRef.current, selectedLogEventNum);
+        const nextLogEventLineNum =
+            getMapKeyByValue(beginLineNumToLogEventNumMapRef.current, selectedLogEventNum + 1);
+
+        if (null === selectedLogEventLineNum) {
+            // logEventNum is not found on the current page.
+            return;
+        }
+
+        let endLineNumber: number;
+        if (null !== nextLogEventLineNum) {
+            endLineNumber = nextLogEventLineNum - 1;
+        } else {
+            // Handle the case when this is the last log event in the file.
+            const model = editorRef.current.getModel();
+            if (null === model) {
+                throw new Error("Unable to get the text model.");
+            }
+            endLineNumber = model.getLineCount() - 1;
+        }
+
+        const selectionRange = new monaco.Range(
+            selectedLogEventLineNum,
+            0,
+            endLineNumber,
+            Infinity
+        );
+
+        editorRef.current.setSelection(selectionRange);
+    }, []);
+
+    // #FixMe: Monaco find indices are always a question mark incorrect on the first
+    //          log event of each page.
+    //      ->The second log event is alway index 1 and so on.
+
+    // Update find action parameters when Zustand store changes
+    const updateFindAction = useCallback(async (
+        selectedLogEventNum:number,
+        beginLineNumToLogEventNumMapRef:React.RefObject<BeginLineNumToLogEventNumMap>
+    ) => {
+        setEditorSelection(selectedLogEventNum, beginLineNumToLogEventNumMapRef);
+        const findAction = editorRef.current.getAction("actions.find");
+        const findWithArgsAction = editorRef.current.getAction("editor.actions.findWithArgs");
+
+        if (findAction && findWithArgsAction) {
+            try {
+                await findAction.run();
+                await findWithArgsAction.run({
+                    searchString: queryString,
+                    isCaseSensitive: queryIsCaseSensitive,
+                    isRegex: queryIsRegex,
+                });
+            } catch (error) {
+                console.error("Error during search:", error);
+            }
+        } else {
+            console.error("Find action or Find with args action is not available.");
+        }
+    }, [queryString,
+        queryIsCaseSensitive,
+        queryIsRegex,
+        setEditorSelection]);
+
+    const closeFind = useCallback(() => {
+        const findController = editorRef.current.getContribution(
+            "editor.contrib.findController"
+        ) as {closeFindWidget: () => void} | null;
+
+        if (findController?.closeFindWidget) {
+            findController.closeFindWidget();
+        }
+    }, []);
+
+    useEffect(() => {
+        closeFind();
+    }, [queryString,
```

Comment:
having a useEffect() hook to call only one function is a bit weird. we can inline the logic
```suggestion
    useEffect(() => {
         const findController = editorRef.current.getContribution(
            "editor.contrib.findController"
        ) as {closeFindWidget: () => void} | null;

        if (findController?.closeFindWidget) {
            findController.closeFindWidget();
        }
    }, [queryString,
```